### DISCUSSION TITLE
Introduce Gradle as build tool

### DISCUSSION THREAD

**Discussion by sauterl** [ppanopticon/vitrivr-ng#105] [Ideas]
What are your thoughts about introducing gradle as build tool on top of npm?

With the [gradle npm plugin](https://github.com/node-gradle/gradle-node-plugin), the build process' dependencies could be reduced to only rely on the gradle wrapper. In contrast to the current system, where nodejs / npm has to be installed.

These pros have fueled this thought:

* In-line with other [vitrivr](https://github.com/vitrivr) components
* Possiblitiy to provide a webjar
* Possiblitiy to automate installation of vitrivr stack

**Comment by silvanheller** [comment]
Are there any cons? Basically, can people still run `ng serve`? does this change something about dependency management or are we still using `package.json`?

**Reply by sauterl** [reply]
I'm not aware of any cons. people **who have npm installed** could still run `ng server` and dependency management would still be handled by `package.json`. Additionally, instead of `ng build --prod`, people would be able to do something along the lines `./gradlew buildFrontend` **without npm installed**, due to the power of gradlewrapper and the aforementioned plugin

**Comment by lucaro** [comment]
The only downside I'd see is that you need to have the dependencies on your disk twice, since gradle just gets its own version of npm which in turn gets all the dependencies. If you set up an environment for development, you'll have all of them in another place anyway. Since nobody needs to use gradle in this case though, this is not a reason not to have it. So 👍 from my side.
